sn_transfers/src/client_transfers/transfer.rs

```diff
@@ -55,13 +42,66 @@ fn select_inputs(
             )
         })?;
 
+    // We need to select the necessary number of dbcs from those that we were passed.
+    let (dbcs_to_spend, change_amount) = select_inputs(available_dbcs, total_output_amount)?;
+
+    let selected_inputs = Inputs {
+        dbcs_to_spend,
+        recipients,
+        change: (change_amount, change_to),
+    };
+
+    create_transfer_with(selected_inputs, reason_hash, None)
+}
+
+/// A function for creating an offline transfer of tokens for a storage payment.
+/// This is done by creating a new network owned output (and a change dbc if any)
+/// by selecting from the available input dbcs, and creating the necessary
+/// spends to do so.
+pub fn create_storage_payment_transfer(
+    available_dbcs: Vec<(Dbc, DerivedKey)>,
+    change_to: PublicAddress,
+    storage_cost: Token,
+    root_hash: Hash,
+    reason_hash: Hash,
+) -> Result<TransferOutputs> {
+    // We need to select the necessary number of dbcs from those that we were passed.
+    let (dbcs_to_spend, change_amount) = select_inputs(available_dbcs, storage_cost)?;
+
+    // We build the recipients to contain just a single output which is for the network owned output.
+    // This is a special output that spendbook peers validating the signed spends (inputs) will be
+    // verifying before accepting them as valid spends for a storage payment. This special output is
+    // expected to be built from hashing: root_hash + input DBCs ids
+    let mut fee_id_bytes = root_hash.slice().to_vec();
+    dbcs_to_spend
+        .iter()
+        .for_each(|(dbc, _)| fee_id_bytes.extend(&dbc.id().to_bytes()));
+
+    let fee = FeeOutput::new(Hash::hash(&fee_id_bytes), storage_cost.as_nano(), root_hash);
+
+    let selected_inputs = Inputs {
+        dbcs_to_spend,
+        recipients: vec![],
+        change: (change_amount, change_to),
+    };
+
+    create_transfer_with(selected_inputs, reason_hash, Some(fee))
+}
+
+/// Select the necessary number of dbcs from those that we were passed.
+fn select_inputs(
+    available_dbcs: Vec<(Dbc, DerivedKey)>,
+    total_output_amount: Token,
+) -> Result<(Vec<(Dbc, DerivedKey)>, Token)> {
+    let mut dbcs_to_spend = Vec::new();
+    let mut total_input_amount = Token::zero();
     let mut change_amount = total_output_amount;
 
     for (dbc, derived_key) in available_dbcs {
         let input_key = dbc.id();
 
-        let dbc_balance = match dbc.revealed_amount(&derived_key) {
-            Ok(revealed_amount) => Token::from_nano(revealed_amount.value()),
+        let dbc_balance = match dbc.token() {
+            Ok(token) => token,
             Err(err) => {
                 warn!("Ignoring input Dbc (id: {input_key:?}) due to not having correct derived key: {err:?}");
```

Comment:
Should we update the warning here, as we might not need to use the derived_key to reveal amounts anymore.